src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -123,13 +141,55 @@ public CopyArtifact(String projectName, String parameters, BuildSelector selecto
         }
 
         this.project = projectName;
+
+        // Apply defaults to all other properties.
+        setParameters(null);
+        setFilter(null);
+        setTarget(null);
+        setExcludes(null);
+        setSelector(null);
+        setFlatten(false);
+        setOptional(false);
+        setFingerprintArtifacts(false);
+    }
+
+    @DataBoundSetter
+    public void setParameters(String parameters) {
         this.parameters = Util.fixEmptyAndTrim(parameters);
-        this.selector = selector;
+    }
+
+    @DataBoundSetter
+    public void setFilter(String filter) {
         this.filter = Util.fixNull(filter).trim();
-        this.excludes = Util.fixNull(excludes).trim();
+    }
+
+    @DataBoundSetter
+    public void setTarget(String target) {
         this.target = Util.fixNull(target).trim();
+    }
+
+    @DataBoundSetter
+    public void setExcludes(String excludes) {
+        this.excludes = Util.fixNull(excludes).trim();
+    }
+
+    @DataBoundSetter
+    public void setSelector(BuildSelector selector) {
+        this.selector = (selector != null ? selector : DEFAULT_BUILD_SELECTOR);
```

Comment:
I guess your backout/rebase lost my earlier comment. In general, any push to a PR which invalidates a portion of the diff will automatically hide any comments made against that earlier diff.

Simplify this setter to assume a `@Nonnull` argument (the one call from the deprecated constructor should then check for null of course) and instead use `DEFAULT_BUILD_SELECTOR` as the field initializer. This is potentially important for [JENKINS-25779](https://issues.jenkins-ci.org/browse/JENKINS-25779).